app/src/main/java/com/example/frolic/QRScanActivity.java

```diff
@@ -0,0 +1,59 @@
+package com.example.frolic;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+
+/**
+ * QRScanActivity displays a QR code and provides a button for scanning QR codes.
+ */
+public class QRScanActivity extends AppCompatActivity {
+
+    private ImageView qrCodeImageView;
+    private Button scanQRCodeButton;
+    private Button btnBack;  // Back button declaration
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.qr_scan);  // Use the provided layout file
+
+        qrCodeImageView = findViewById(R.id.ivQRCode);
+        scanQRCodeButton = findViewById(R.id.btnScanQRCode);
+        btnBack = findViewById(R.id.btnBack);  // Initialize the back button
+
+        // Set up the back button click listener
+        btnBack.setOnClickListener(v -> {
+            Intent intent = new Intent(QRScanActivity.this, EntrantDashboardActivity.class);
+            startActivity(intent);
+            finish();
+        });
+
+        // Retrieve event data from the Intent
+        String eventData = getIntent().getStringExtra("eventData");
```

Comment:
I'm not sure what "eventData" means here, but the way to get the QR code for a certain event is by calling QRCodeGenerator.generateQRCode() on the qrHash field in the events collection in Firebase. So if eventData is the qrHash then its good.

Reply:
Yes eventData is the qrhash